library/src/main/java/io/github/wax911/library/annotation/processor/fragment/FragmentAnalyzer.kt

```diff
@@ -0,0 +1,9 @@
+package io.github.wax911.library.annotation.processor.fragment
+
+/**
+ * A contract for something that can parse a query and provide a full analysis of what fragments are referenced, and
+ * whether the fragments are defined with the query.
+ */
+interface FragmentAnalyzer {
```

Comment:
In talking with the owners of this Anitrend library, they agreed with me that for now we should just keep it simple (and use regular expressions). In the future, they may want to actually parse queries into structural data. So I'd like to leave them the option to provide that kind of implementation.